File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/FieldEntity.java
package com.github.ol_loginov.heaplibweb.repository.heap;

import com.github.ol_loginov.heaplibweb.repository.EntityIdentity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.*;

@Entity
@Table(name = "Field")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class FieldEntity extends EntityIdentity {
	private int heapId;
	/**
	 * @see JavaClassEntity
	 */
	private long declaringClassId;
	@Column(nullable = false)
	private String name;
	private boolean staticFlag;
	/**
	 * Might be JavaClassEntity or internal primitive id
	 */
	private int typeId;

	@Override
	public String toString() {
		return idString() + " (class=" + declaringClassId + ",name=" + name + ")";
	}
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/FieldEntity.kt
package com.github.ol_loginov.heaplibweb.repository.heap

import com.github.ol_loginov.heaplibweb.hprof.HprofValueType
import com.github.ol_loginov.heaplibweb.repository.EntityIdentity
import org.springframework.lang.NonNull

class FieldEntity private constructor(
    id: Int,
    var declaringClassId: Long,
    var nameId: Int,
    var staticFlag: Boolean,
    var typeTag: Byte
) : EntityIdentity(id) {
    constructor(declaringClassId: Long, nameId: Int, staticFlag: Boolean, typeTag: Byte) : this(0, declaringClassId, nameId, staticFlag, typeTag)
    private constructor() : this(0, 0, 0, false, 0)

    val type
        get() = HprofValueType.valueOf(typeTag.toUByte())

    @NonNull
    override fun toString() = "${super.idString()} (class=$declaringClassId,name=$nameId)"
}